### PR TITLE
Don't draw a marker if last hour average outside of scope

### PR DESCRIPTION
**Pull request by AkkarinForest**

### CODE REVIEW

File: app/assets/javascripts/tests/_draw_session.test.js [hunk 1]
```diff
@@ -31,8 +31,40 @@ test('drawMobileSessionStartPoint calls drawMarker', t => {
...
+  const sensors = { anySelected: () => true , tmpSelected: () => false }
+  const heat = { getLevel: () => 1 }
+  const drawSessionMock = _drawSession({ map, sensors, heat });
+
+  const session = { last_hour_average: 1 };
+  const actual = drawSessionMock.drawFixedSession(session);
+
+  const expected = [marker]
+
+  t.deepEqual(actual, expected  );
```
Comment (3v0k4) [inline app/assets/javascripts/tests/_draw_session.test.js]: `  ` spaces

File: app/assets/javascripts/tests/_draw_session.test.js [hunk 2]
```diff
@@ -31,8 +31,40 @@ test('drawMobileSessionStartPoint calls drawMarker', t => {
...
+  const map = { drawMarker: () => marker };
+  const sensors = { anySelected: () => true , tmpSelected: () => false }
+  const heat = { getLevel: () => 1 }
+  const drawSessionMock = _drawSession({ map, sensors, heat });
+
+  const session = { last_hour_average: 1 };
+  const actual = drawSessionMock.drawFixedSession(session);
+
+  const expected = [marker]
+
```
Comment (3v0k4) [inline app/assets/javascripts/tests/_draw_session.test.js]: I guess you can rm this line

File: app/assets/javascripts/tests/_draw_session.test.js [hunk 3]
```diff
@@ -31,8 +31,40 @@ test('drawMobileSessionStartPoint calls drawMarker', t => {
...
 
-const _drawSession = ({ map }) => {
+test('drawFixedSession creates a marker if heat level is not null', t => {
+  const marker = {};
+  const map = { drawMarker: () => marker };
+  const sensors = { anySelected: () => true , tmpSelected: () => false }
+  const heat = { getLevel: () => 1 }
+  const drawSessionMock = _drawSession({ map, sensors, heat });
+
+  const session = { last_hour_average: 1 };
```
Comment (3v0k4) [inline app/assets/javascripts/tests/_draw_session.test.js]: This line should prolly be in the block above

File: app/assets/javascripts/tests/_draw_session.test.js [hunk 4]
```diff
@@ -31,8 +31,40 @@ test('drawMobileSessionStartPoint calls drawMarker', t => {
   t.end();
 });
 
-const _drawSession = ({ map }) => {
+test('drawFixedSession creates a marker if heat level is not null', t => {
```
Comment (3v0k4) [inline app/assets/javascripts/tests/_draw_session.test.js]: Seems like you could move some noise in the builder (`drawSessionMock`) cause it's a bit noisy here

File: app/assets/javascripts/code/services/_draw_session.js [hunk 5]
```diff
@@ -50,6 +50,7 @@ export const drawSession = (
 
       if (sensors.anySelected() && !sensors.tmpSelected() && session.last_hour_average) {
         level = calculateHeatLevel(heat, session.last_hour_average);
+        if (!level) { return  session.markers };
```
Comment (3v0k4) [inline app/assets/javascripts/code/services/_draw_session.js]: You can remove `{}`. The nested if is a bit confusing maybe we can refactor around?

File: app/assets/javascripts/tests/_draw_session.test.js [hunk 6]
```diff
@@ -31,8 +31,35 @@ test('drawMobileSessionStartPoint calls drawMarker', t => {
...
+  const actual = drawSessionMock.drawFixedSession(session);
+
+  const expected = [marker]
+  t.deepEqual(actual, expected);
+
+  t.end();
+})
+
+test('drawFixedSession does not create a marker if heat level is null', t => {
+  const drawSessionMock = _drawSession({});
```
Comment (3v0k4) [inline app/assets/javascripts/tests/_draw_session.test.js]: Have you considered passing something like `const heat = { getLevel: () => null }`. Right now the test relies on the default which is hidden in `_drawSession`.

Comment (3v0k4) [reply app/assets/javascripts/tests/_draw_session.test.js]: Also, you may want to call it stub. Mock has another meaning. Maybe just service would do (we have been calling it service in other tests).

Stub is a test double that is instructed to return specific values.
Mock is a test double that performs expectations on calls.

File: app/assets/javascripts/code/services/_heat.js [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+      } else if (value < this.getValue("highest")) {
+        return 4;
+      } else {
+        return null;
+      }
+    },
+
+    outsideOfScope: function(value) {
+      return (value < this.getValue("lowest")) || (value > this.getValue("highest"))
+    }
```
Comment (AkkarinForest) [inline app/assets/javascripts/code/services/_heat.js]: that is the only new function I added, the rest is just copied from the old heat.js

**Pull request closed without merging**